### PR TITLE
Ruby OTel lambda wrapper

### PR DESCRIPTION
**Pull request by kubawach**
Ruby OTel lambda wrapper.

Main wrapper file - `ruby/src/wrapper.rb`

Caches `google-protobuf-3.19.3-x86_64-linux` gem under `ruby/cached-gems` as platform-dependent gem needed by the AWS lambda runtime.

### CODE REVIEW

File: ruby/build-layer.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+cp src/*.rb build/ruby/lib
+src=$?
+
+echo "Packaging dependencies into zip archive..."
+cd build
+zip -qr9 ruby.zip *
+zip=$?
+
+if [[ "$ruby" -ne 0 || "$copy" -ne 0 || "$scripts" -ne 0 || "$src" -ne 0 || "$zip" -ne 0 ]] ; then
+  echo "Could not build layer"; exit 1
```
Comment (mateuszrzeszutek) [inline ruby/build-layer.sh:37]: Nit: why not just call `set -e` at the beginning of the script?

File: ruby/src/wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+end
+
+class ServerTimingHeader
+
+  FALSE_VALUES = %w(false no 0).freeze
+
+  def is_enabled?
+    value = ENV.fetch('SPLUNK_TRACE_RESPONSE_HEADER_ENABLED', 'true').to_s.downcase
+    return false if FALSE_VALUES.include?(value)
+    return true
```
Comment (mateuszrzeszutek) [inline ruby/src/wrapper.rb:18]:
```suggestion
    return !FALSE_VALUES.include?(value)
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: ruby/src/wrapper.rb (hunk 2 above)
Comment (kubawach) [reply ruby/src/wrapper.rb]: Yeah, that makes sense :)

File: ruby/build-layer.sh (hunk 1 above)
Comment (kubawach) [reply ruby/build-layer.sh]: +1

**Pull request merged**